packages/react/README.md

```diff
@@ -5,13 +5,579 @@
+| `deleteDocument`    | Delete document entirely (draft and published) |
+| `discardDraft`      | Discard draft changes, revert to published     |
+
+#### Creating Documents
+
+To create a document, you must:
+
+1. Generate your own document ID (using `crypto.randomUUID()`)
+2. Create a document handle with `createDocumentHandle`
+3. Apply the `createDocument` action with initial data
+
+```tsx
+import {useApplyDocumentActions, createDocumentHandle, createDocument} from '@sanity/sdk-react'
+
+function CreateArticleButton() {
+  const apply = useApplyDocumentActions()
+
+  const handleCreate = () => {
+    const newId = crypto.randomUUID()
+    const handle = createDocumentHandle({
+      documentId: newId,
+      documentType: 'article',
+    })
+
+    apply(
+      createDocument(handle, {
+        title: 'New Article',
+        status: 'draft',
+        author: {_type: 'reference', _ref: 'author-123'},
+      }),
+    )
+
+    // Navigate to the new document
+    navigate(`/articles/${newId}`)
+  }
+
+  return <button onClick={handleCreate}>Create Article</button>
+}
+```
+
+#### Publishing Documents
+
+```tsx
+import {useApplyDocumentActions, publishDocument, useDocument} from '@sanity/sdk-react'
+
+function PublishButton({handle}: {handle: DocumentHandle}) {
+  const apply = useApplyDocumentActions()
+  const {data: doc} = useDocument(handle)
+
+  // Check if document has unpublished changes (is a draft)
+  const isDraft = doc?._id?.startsWith('drafts.')
+
+  return (
+    <button disabled={!isDraft} onClick={() => apply(publishDocument(handle))}>
+      Publish
+    </button>
+  )
+}
+```
+
+#### Deleting Documents
+
+```tsx
+import {useApplyDocumentActions, deleteDocument} from '@sanity/sdk-react'
+
+function DeleteButton({handle}: {handle: DocumentHandle}) {
+  const apply = useApplyDocumentActions()
+
+  const handleDelete = () => {
+    if (confirm('Are you sure?')) {
+      apply(deleteDocument(handle))
+    }
+  }
+
+  return <button onClick={handleDelete}>Delete</button>
+}
+```
+
+#### Batch Operations
+
+Apply multiple actions as a single transaction:
+
+```tsx
+const apply = useApplyDocumentActions()
+
+// Create and immediately publish
+const newHandle = createDocumentHandle({
+  documentId: crypto.randomUUID(),
+  documentType: 'article',
+})
+
+apply([createDocument(newHandle, {title: 'Breaking News'}), publishDocument(newHandle)])
+
+// Publish multiple documents at once
+apply([publishDocument(handle1), publishDocument(handle2), publishDocument(handle3)])
+```
+
+---
+
+### Suspense Pattern
+
+All data hooks use React Suspense. Wrap components that fetch:
+
+```tsx
+function App() {
+  return (
+    <Suspense fallback={<Skeleton />}>
+      <ArticleList />
+    </Suspense>
+  )
+}
+
+function ArticleList() {
+  const {data: articles} = useDocuments({documentType: 'article'})
+
+  return (
+    <ul>
+      {articles.map((handle) => (
+        <Suspense key={handle.documentId} fallback={<li>Loading...</li>}>
```

Comment:
```suggestion
      {articles.map((handle) => (
        {/* Wrap each list item in its own Suspense boundary to prevent full list re-renders when one item updates */}
        <Suspense key={handle.documentId} fallback={<li>Loading...</li>}>
```